orm/index.go

```diff
@@ -118,16 +123,35 @@ func (i Index) Update(db weave.KVStore, prev Object, save Object) error {
 }
 
 // GetLike calculates the index for the given pattern, and
-// returns a list of all pk that match (may be empty), or an error
+// returns a list of all pk that match (may be nil when empty), or an error
 func (i Index) GetLike(db weave.ReadOnlyKVStore, pattern Object) ([][]byte, error) {
 	indexes, err := i.index(pattern)
 	if err != nil {
 		return nil, err
 	}
+	var r [][]byte
 	for _, index := range indexes {
-		return i.GetAt(db, index)
+		pks, err := i.GetAt(db, index)
+		if err != nil {
+			return nil, err
+		}
+		if i.unique {
+			return pks, nil
+		}
+		r = append(r, pks...)
+	}
+	return deduplicate(r), nil
+}
+
+func deduplicate(r [][]byte) [][]byte {
+	for i, v := range r {
+		for j := i + 1; j < len(r); j++ {
+			if bytes.Equal(v, r[j]) {
+				r = append(r[0:j], r[j+1:]...)
```

Comment:
is it safe to re-assign r while in the range loop?

code looks good, I would have a separate `res` variable, but I do see the code is smaller here, even if a bit confusing modifying a loop guard in the loop

Reply:
I remembered it was safe unlike java but when I searched for some resources on the net I am not sure anymore. It definitively was not save with older versions of Go in <=2017. Anyway, I refactored into a classic for loop to be on the save side. 
fyi: I found this wiki entry which has a nice trick to reuse memory with append: https://github.com/golang/go/wiki/SliceTricks#filtering-without-allocating